### ISSUE TITLE
Enforce whitespace consistency

### ISSUE THREAD

**Issue by scouten** [FuelLabs/cargo-toml-lint#10]
I'm looking for a linter that enforces whitespace rules. Some things I'd like to enforce via automation:

* `=` as a token is always surrounded by exactly one space
* `[...]` as an inline array syntax has no spaces inside the brackets
* `{...}` as an inline table syntax has exactly one space inside the brackets
* No trailing whitespace on any line.
* No leading whitespace except if the line starts with a comment.
* Exactly one newline character at end of file. (See also #6.)

Would you be open to a contribution that adds these rules? (If yes, I'll start looking into it.)

**Comment by adlerjohn** [comment]
> Would you be open to a contribution that adds these rules?

Absolutely, and they would be very appreciated! In order to keep review times down, I suggest making a PR for each of them individually instead of all at once.
